src/dune_pkg/dev_tool.ml

```diff
@@ -0,0 +1,44 @@
+open Import
+
+type t =
+  { name : Package_name.t
+  ; dependencies : Package_dependency.t list
+  }
+
+let to_local (t : t) : Local_package.t =
+  let open Local_package in
+  { name = t.name
+  ; version = None
+  ; dependencies = t.dependencies
+  ; conflicts = []
+  ; depopts = []
+  ; pins = Package_name.Map.empty
+  ; conflict_class = []
+  ; loc = Loc.none
+  }
+;;
+
+let lock_dir = Path.Source.of_string "dev_tools.locks"
```

Comment:
For the source directory the answer is here https://github.com/ocaml/dune/pull/10647#issuecomment-2206102515.

What about `dev_tools/ocamlformat.lock` ?

Reply:
Ok, we discussed this with the team and generally the consensus was that putting it in the source folder is fine.

Personally I try to avoid underscores so I'd probably name it `devtool.locks` or `dev-tool.locks` but the exact name isn't all that important at this stage as it can easily be changed.

Reply:
Is the expectation that this file is going to be committed in source control?

Reply:
I don't really get the question but if it's about `dev_tool.ml`, that was just a starting point, if its content could move to other modules, we remove it. Example `lock_dir` here will end up in `Lock_dir` module.